File: Haskell/beginner/1174.hs
-- https://judge.beecrowd.com/pt/problems/view/1174
-- Seleção em Vetor I

-- Importing forM to iterate with IO effects
import Control.Monad (forM_)
import Numeric (showFFloat)

-- Function to format float values to one decimal place
showFFloat1 :: Float -> String
showFFloat1 x = showFFloat (Just 1) x ""


-- Main function
main :: IO ()
main = do

    -- Read the first 100 float values into a list
    usrInputs <- sequence [readLn :: IO Float | _ <- [0..99]]

    -- Iterate over the list with index and print the values that are less than or equal to 10
    forM_ (zip [0..] usrInputs) $ \(i, val) ->
        -- Checking if the value is less than or equal to 10
        if val <= 10
            -- Attention to the format of the output
            then putStrLn $ "A[" ++ show i ++ "] = " ++ showFFloat1 val
            else return ()

-- By: Fernando Serra
-- https://github.com/fernandosserra

File: Python/beginner/1174.py
# https://judge.beecrowd.com/pt/problems/view/1174
# Seleção em Vetor I

# Read the first 100 floating-point numbers
usrInputs = [float(input()) for _ in range(100)]

# Iterate through the list and print the index and value of elements less than 10
for i, value in enumerate(usrInputs):
    if value <= 10:
        print(f"A[{i}] = {value:.1f}")
# The format specifier .1f ensures that the output is formatted to one decimal place


# By: Fernando Serra
# https://github.com/fernandosserra